### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by CleanCut**
Manual changes:
- ~Update Rust from 1.77 to 1.83 (Is there any reason not to use the latest stable version of Rust?)~ Reverted pending decision in on what MSRV policy to have
- Update `http` from 1.0 to 1.2, resolves #137 
- Update `tokio` from 1.41 to 1.42, resolves #139 
- Silenced an unused code warning in the example

Changes to `Cargo.lock` from running `cargo update`:
- Update `prost` from 0.13.3 to 0.13.4, resolves #136 
- Update `prost-build` from 0.13.3 to 0.13.4, resolves #135 
- Update `thiserror` from 2.0.3 to 2.0.6, resolves #138

### CODE REVIEW

**Comment by jorendorff** [comment]
> Update Rust from 1.77 to 1.83 (Is there any reason not to use the latest stable version of Rust?)

Bumping the Rust version effectively changes our MSRV. Although older versions would not instantly break, it definitely changes the minimum Rust version we _test_ with.

Really we should have an explicit MSRV policy of some sort (even if it's "we only support the latest"), and [set the `rust-version` property in crates/*/Cargo.toml](https://doc.rust-lang.org/cargo/reference/rust-version.html) accordingly. Would you mind giving that a shot?

File: rust-toolchain.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [toolchain]
-channel = "1.77.0"
+channel = "1.83.0"
```
Comment (jorendorff) [inline rust-toolchain.toml:2]: See comment in the PR.

**Review by jorendorff — APPROVED** [review]
Thanks! Approving conditionally:

The dependency updates alone are fine, if tests still pass with the old Rust version (that is, back out the `version = 4` change in Cargo.lock and the rust-toolchain.toml change).

Or, document any reasonable MSRV policy in the README and set `[toolchain] channel` in rust-toolchain.toml and `[package] rust-version` in the Cargo.toml files accordingly. If the MSRV policy covers more than just the latest Rust release, ideally our CI should run the tests in both the oldest and the newest versions of Rust we claim to support.

**Comment by CleanCut** [comment]
I backed out the Rust version change and started a [discussion about what MSRV policy to implement](https://github.com/github/blackbird/issues/8711).

**Pull request merged**